### PR TITLE
feat: github action to fetch metrics per space

### LINKED ISSUE #116: Add user facing aggregations
> Part of https://github.com/web3-storage/w3infra/issues/117 to track user facing aggregations

Now that we have UCAN STREAM log in action with Kinesis stream, in order to easily keep track of user facing aggregations.

### PR DESCRIPTION
**Pull request by vasco-santos**
Adds github action to query space metrics for given space

```
npm run fetch-metrics-for-space

> w3infra@0.0.0 fetch-metrics-for-space
> npm run fetch-metrics-for-space -w tools


> tools@1.0.0 fetch-metrics-for-space
> node cli.js fetch-metrics-for-space

Metrics found for provided space DID: 1
upload/add-total: 7
```

TODO
- [x] Add AWS secrets

Closes #116

### CODE REVIEW

File: tools/fetch-metrics-for-space.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+ * 
+ * @param {string} name 
+ * @returns {string}
+ */
+function mustGetEnv (name) {
+  if (!process.env[name]) {
+    throw new Error(`Missing env var: ${name}`)
+  }
+
+  // @ts-expect-error there will always be a string there, but typescript does not believe
```
Comment (alanshaw) [inline tools/fetch-metrics-for-space.js:71]: I think if you assign to a variable and then do the check it'll not complain...

File: tools/fetch-metrics-for-space.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+  return 'us-west-2'
+}
+
+/**
+ * @param {string} tableName
+ * @param {string} env
+ * @param {string} region
+ */
+function getDynamoDb (tableName, env, region) {
```
Comment (alanshaw) [inline tools/fetch-metrics-for-space.js:91]: How does it authenticate?

Comment (vasco-santos) [reply tools/fetch-metrics-for-space.js:91]: Do you mean authenticate to run the action? This is only for admin usage at the moment. We will integrate this with API later on (with `space/info`, or other API) where users will be able to interact with the system. Github triggers can only be run by collaborators with run permissions in repo.

Or do you mean dynamoDB? Yeah, I guess we need to put AWS credentials in... In my env I have them configured, and seed.run (where integration tests run with similar setup) also should have them set. I will add them

Comment (alanshaw) [reply tools/fetch-metrics-for-space.js:91]: yes the latter!

**Comment by vasco-santos** [comment]
Now node18 is failing to download 😡

**Comment by vasco-santos** [comment]
@alanshaw looks like node installer is back :)

File: tools/fetch-metrics-for-space.js (hunk 1 above)
Comment (alanshaw) [reply tools/fetch-metrics-for-space.js:71]: No like:

```js
function mustGetEnv (name) {
  const value = process.env[name]
  if (!value) throw new Error(`Missing env var: ${name}`)
  return value
}
```

**Review by alanshaw — APPROVED** [review]

**Pull request merged**